src/core/qgspointlocator.cpp

```diff
@@ -685,7 +694,8 @@ bool QgsPointLocator::rebuildIndex( int maxFeaturesToIndex )
     return true; // nothing to index
 
   QgsFeatureRequest request;
-  request.setSubsetOfAttributes( QgsAttributeList() );
+  //request.setSubsetOfAttributes( QgsAttributeList() );
+  request.setFilterFids( mFeatureIds );
```

Comment:
This should be:

    if ( !mFeatureIds.empty() )
        request.setFilterFids( mFeatureIds );

(I think ideally QgsFeatureRequest should ignore empty feature lists, but at the moment it still sets the filter type to FilterFids even if an empty list is given -- that would need careful testing though, so go with the easy option here).

Reply:
done with an else for request.setSubsetOfAttributes